inputFiles/compositionalMultiphaseFlow/overallCompositionFormulation/co2_2d_plume/co2_2d_plume_Z.xml

```diff
@@ -0,0 +1,233 @@
+      target="/Tasks/compflowStatistics" />
+
+    <PeriodicEvent
+      name="outputs"
+      timeFrequency="1e6"
+      targetExactTimestep="1"
+      target="/Outputs/vtkOutput" />
+
+    <PeriodicEvent
+      name="restarts"
+      timeFrequency="1e6"
+      targetExactTimestep="1"
+      target="/Outputs/sidreRestart"
+    />    
+
+  </Events>
+
+  <NumericalMethods>
+    <FiniteVolume>
+      <TwoPointFluxApproximation
+        name="fluidTPFA"/>
+    </FiniteVolume>
+  </NumericalMethods>
+
+  <ElementRegions>
+    <CellElementRegion
+      name="region"
+      cellBlocks="{ cellBlock }"
+      materialList="{ fluid, rock, relperm }"/>
+  </ElementRegions>
+
+  <Constitutive>
+    <CO2BrinePhillipsFluid
+      name="fluid"
+      logLevel="1"
+      phaseNames="{ gas, water }"
+      componentNames="{ co2, water }"
+      componentMolarWeight="{ 44e-3, 18e-3 }"
+      phasePVTParaFiles="{ pvtgas.txt, pvtliquid.txt }"
+      flashModelParaFile="co2flash.txt"/>
+
+    <CompressibleSolidConstantPermeability
+      name="rock"
+      solidModelName="nullSolid"
+      porosityModelName="rockPorosity"
+      permeabilityModelName="rockPerm"/>
+    <NullModel
+      name="nullSolid"/>
+    <PressurePorosity
+      name="rockPorosity"
+      defaultReferencePorosity="0.2"
+      referencePressure="1.1e7"
+      compressibility="1.0e-9" />
+    <ConstantPermeability
+      name="rockPerm"
+      permeabilityComponents="{ 1.0e-13, 1.0e-13, 1.0e-14 }"/>
+
+    <TableRelativePermeability
+      name="relperm"
+      phaseNames="{ gas, water }"
+      wettingNonWettingRelPermTableNames="{ waterRelativePermeabilityTable,
+                                            gasRelativePermeabilityTable }"/>
+  </Constitutive>
+
+  <FieldSpecifications>
+    <!-- Initial conditions  -->
+    <FieldSpecification
+      name="initial_pressure"
+      setNames="{ all }"
+      objectPath="ElementRegions/region"
+      fieldName="pressure"
+      initialCondition="1"
+      logLevel="2"
+      scale="1.1e6" />
+    <FieldSpecification
+      name="initial_temperature"
+      setNames="{ all }"
+      objectPath="ElementRegions/region"
+      fieldName="temperature"
+      initialCondition="1"
+      scale="368.15" />
+    <FieldSpecification
+      name="initial_co2"
+      setNames="{ all }"
+      objectPath="ElementRegions/region"
+      fieldName="globalCompFraction"
+      initialCondition="1"
+      component="0"
+      scale="0.0" />
+    <FieldSpecification
+      name="initial_h2o"
+      setNames="{ all }"
+      objectPath="ElementRegions/region"
+      fieldName="globalCompFraction"
+      initialCondition="1"
+      component="1"
+      scale="1.0" />
+
+    <!-- Injection rate: 0.002 kg/s -->
+    <SourceFlux
+      name="sourceTerm"
+      objectPath="ElementRegions/region/cellBlock"
+      component="0"
+      scale="-0.002"
+      setNames="{ source }"/>
+
+    <!-- Boundary conditions at right end  -->
+    <FieldSpecification
```

Comment:
These boundary conditions are applied on the faces. Is there a different code path if we impose a Dirichlet bc on actual elements instead of faces?